src/User/Access/Gate.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\User\Access;
+
+use Flarum\Database\AbstractModel;
+use Flarum\Event\GetPermission;
+use Flarum\User\User;
+use Illuminate\Contracts\Events\Dispatcher;
+use Illuminate\Support\Arr;
+
+class Gate
+{
+    protected const EVALUATION_CRITERIA = [
+        AbstractPolicy::FORCE_DENY => false,
```

Comment:
I think a comment here should say that the order of this array is of the highest importance since it's the order they will be evaluated in.

Maybe the const could even be named `EVALUATION_CRITERIA_PRIORITY` or similar?